core/kernel/impl/class.ApiModelOO.php

```diff
@@ -133,58 +120,44 @@ public function getResourceDescriptionXML($uriResource)
                 $object     = trim($row['object']);
                 $lang       = trim($row['l_language']);
 
-                $nodeName = null;
+                if (strpos($predicate, '#') === false) {
+                    continue;
+                }
 
-                foreach ($allNs as $namespaceId => $namespaceUri) {
-                    if ($namespaceId === 'xml:base') {
-                        continue;
-                    }
-                    if (preg_match("/^" . preg_quote($namespaceUri, '/') . "/", $predicate)) {
-                        if (!array_key_exists($namespaceId, $currentNs)) {
-                            $currentNs[$namespaceId] = $namespaceUri;
-                            $root->setAttribute($namespaceId, $namespaceUri);
-                        }
-                        $nodeName = str_replace('xmlns:', '', $namespaceId) . ':' . str_replace($namespaceUri, '', $predicate);
-                        break;
-                    }
+                [$namespace, $property] = explode('#', $predicate, 2);
+                $namespace = "$namespace#";
+
+                if (!isset($namespaces[$namespace])) {
+                    $namespaceId            = sprintf('ns%u', ++$namespaceCounter);
+                    $namespaces[$namespace] = $namespaceId;
+                    $root->setAttribute("xmlns:$namespaceId", $namespace);
                 }
 
-                $resourceValue = false;
-                foreach ($allNs as $namespaceUri) {
-                    if (
-                        preg_match('/^' . preg_quote($namespaceUri, '/') . '/', $object) ||
-                        preg_match("/^http:\/\/(.*)#[a-zA-Z1-9]*/", $object)
-                    ) {
-                        $resourceValue = true;
-                        break;
+                $namespaceId = $namespaces[$namespace];
+                $nodeName    = "$namespaceId:$property";
```

Comment:
More a matter of taste, but, could I ask you to inject variables into strings in the same way everywhere, please? :upside_down_face: 
Because on line 101 it's injected with `...{$var}`; on line 110 (and further) "...:$var". This shouldn't change anything from any of the perspectives, but it will improve readability a bit (at least for me - as my eyes are just stumbling on those lines)

Reply:
I'm gonna unify them, but removing the curly braces, if you don't mind :)

Reply:
Nah, don't mind at all :) Thanks!